packages/interface/src/components/action-item/index.js

```diff
@@ -1,27 +1,37 @@
 /**
  * External dependencies
  */
-import { isEmpty, noop } from 'lodash';
+import { isArray, isEmpty, noop } from 'lodash';
 
 /**
  * WordPress dependencies
  */
 import { ButtonGroup, Button, Slot, Fill } from '@wordpress/components';
+import deprecated from '@wordpress/deprecated';
 import { Children } from '@wordpress/element';
 
 function ActionItemSlot( {
 	name,
-	as = [ ButtonGroup, Button ],
+	as: Component = ButtonGroup,
 	fillProps = {},
 	bubblesVirtually,
 	...props
 } ) {
-	const [ Container, Item ] = as;
+	if ( isArray( Component ) ) {
+		deprecated(
+			'Passing a tuple of components with `as` prop to `ActionItem.Slot` component',
+			{
+				alternative: 'a component with `as` prop',
+				version: '10.3',
+			}
+		);
+		Component = Component[ 0 ];
+	}
 	return (
 		<Slot
 			name={ name }
 			bubblesVirtually={ bubblesVirtually }
-			fillProps={ { as: Item, ...fillProps } }
+			fillProps={ fillProps }
```

Comment:
Regarding the [above comment](https://github.com/WordPress/gutenberg/pull/29340/files#r583457652) I meant here:
`as: Item` to be `Component[ 1 ]` if is an array.

Reply:
It defaults to Button, so I guess it should be fine. Technically it will still work. If you think, we should provide a backward compatibility solution, I can do it.

Reply:
With `Button` should be just fine yes - also I trust you a lot in back compat things :)

Reply:
I would be more concerned it wasn't an internal package that isn't exposed in the public API of WordPress 😄